worker/error-handler/index.js

```diff
@@ -0,0 +1,119 @@
+import amqp from 'amqplib';
+
+const workerQueue = process.env.WORKERQUEUE;
+const resultQueue = process.env.RESULTSQUEUE;
+const rabbitHost = process.env.RABBITHOST;
+const rabbitUser = process.env.RABBITUSER;
+const rabbitPass = process.env.RABBITPASS;
+
+let channel;
+
+/**
+ * This worker handles errors that may appear in the specific workflows the dispatcher handles.
+ * If a worker throws an error, this worker will pick up that message / job information
+ * from the dead letter queue in order to trigger mail, chat messages and cleanups.
+ */
+export async function initialize(
+  rabbitHost,
+  rabbitUser,
+  rabbitPass,
+  workerQueue,
+  resultQueue
+) {
+  const connection = await amqp.connect({
+    hostname: rabbitHost,
+    username: rabbitUser,
+    password: rabbitPass,
+    heartbeat: 60
+  });
+  channel = await connection.createChannel();
+
+  channel.assertQueue(workerQueue, {
+    durable: true
+  });
+
+  console.log(`Worker waiting for messages in ${workerQueue}.`);
+  channel.consume(
+    workerQueue,
+    async function (msg) {
+      try {
+        const job = JSON.parse(msg.content.toString());
+        console.log('Received a failed job in dead letter queue ...');
+
+        // Handling different recipients based on failed instance
+        // External failure: failures in validators -> mail to original job sender; POST message to chat
+        // Internal failure: failures after validators -> POST message to chat
+        const failedTask = job.nextTask?.task?.type;
+        let recipients;
+        if (failedTask && failedTask.toLowerCase().indexOf('validator') > -1) {
```

Comment:
hm... takes me quite some time to understand this line. Could make a little explanation?